MushroomObserver/mushroom-observer
# `observation_form_map.js` - jQuery > ES6 class, add system test

Refactor is prep for making this a simpler Stimulus controller.

### System Test ###
tests map load only, because i don't know how to test what goes on inside the gmaps iframe.

### Manual Test ###
- Create a new obs. 
- Enter a location
- Click "Locate on Map"
- Map should come up with the location
- Click to make a point marker
- Lat/Lng/Elv fields should populate with the data for the point
- Click the "Clear Map" button to remove the marker
- Lat/Lng/Elv fields should be clear too
